nni/mutable/symbol.py

```diff
@@ -228,6 +229,76 @@ def condition(pred: Any, true: Any, false: Any) -> SymbolicExpression | Any:
             return symbol_obj.expr_cls(lambda t, c, f: t if c else f, '{} if {} else {}', [true, pred, false])
         return true if pred else false
 
+    @symbolic_staticmethod
+    def case(pred_expr_pairs: list[tuple[Any, Any]]) -> SymbolicExpression | Any:  # type: ignore
+        """Return the first expression with predicate that is true.
+
+        For example::
+
+            if (x < y) return 17;
+            else if (x > z) return 23;
+            else (y > z) return 31;
+
+        Equivalent to::
+
+            SymbolicExpression.case([(x < y, 17), (x > z, 23), (y > z, 31)])
+        """
+
+        def _case_fn(*pred_expr_pairs):
+            assert len(pred_expr_pairs) % 2 == 0
+            for pred, expr in zip(pred_expr_pairs[::2], pred_expr_pairs[1::2]):
+                if pred:
+                    return expr
+            raise RuntimeError('No matching case')
+
+        chained_pairs = list(itertools.chain(*pred_expr_pairs))
+        symbol_obj = first_symbolic_object(*chained_pairs)
+        if symbol_obj is not None:
+            return symbol_obj.expr_cls(
+                _case_fn,
+                'case([' + ', '.join(['({}, {})'] * len(pred_expr_pairs)) + '])',
+                chained_pairs
+            )
+        return _case_fn(*chained_pairs)
+
+    @symbolic_staticmethod
+    def switch_case(branch: Any, expressions: dict[Any, Any]) -> SymbolicExpression | Any:
+        """Select the expression that matches the branch.
+
+        C-style switch:
+
+        .. code-block:: cpp
+
+            switch (branch) {  // c-style switch
+                case 0: return 17;
+                case 1: return 31;
+            }
+
+        Equivalent to::
+
+            SymbolicExpression.switch_case(branch, {0: 17, 1: 31})
```

Comment:
So fancy...

Reply:
I think it's actually simpler than nested `if-else`.

Reply:
I mean, `case` + `switch_case` is a little bit too fancy to me.
I have just checked the Pythonic way, `match` statement, but unluckily it's even more fancy...  :(

Reply:
One of them is needed in symbolic shape inference. The other one is used in nn-meter profiler.

Tensorflow have both [switch_case](https://www.tensorflow.org/api_docs/python/tf/switch_case) and [case](https://www.tensorflow.org/api_docs/python/tf/case).